modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/interceptors/BasicAuthInterceptor.kt

```diff
@@ -0,0 +1,36 @@
+package au.com.shiftyjelly.pocketcasts.servers.interceptors
+
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.Feature
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.FeatureFlag
+import java.net.HttpURLConnection
+import okhttp3.Credentials
+import okhttp3.Interceptor
+import okhttp3.Response
+
+internal class BasicAuthInterceptor : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val originalRequest = chain.request()
+        val response = chain.proceed(originalRequest)
+
+        if (!FeatureFlag.isEnabled(Feature.BASIC_AUTHENTICATION) || response.code != HttpURLConnection.HTTP_UNAUTHORIZED) {
+            return response
+        }
+
+        val header = response.header("WWW-Authenticate")
+        if (header == null || !header.contains("Basic")) {
+            return response
+        }
+
+        val httpUrl = originalRequest.url
+        if (httpUrl.username.isEmpty() || httpUrl.password.isEmpty()) {
+            return response
+        }
+
+        response.close()
+        val credentials = Credentials.basic(httpUrl.username, httpUrl.password)
+        val newRequest = originalRequest.newBuilder()
+            .header("Authorization", credentials)
+            .build()
```

Comment:
Perhaps we should always do this with the initial request if credentials are present in a URL? I think it is reasonable to assume that if they are present they are expected by the server.